pkg/block/fetcher.go

```diff
@@ -401,14 +401,26 @@ func NewLabelShardedMetaFilter(relabelConfig []*relabel.Config) *LabelShardedMet
 	return &LabelShardedMetaFilter{relabelConfig: relabelConfig}
 }
 
-// Filter filters out blocks that filters blocks that have no labels after relabelling.
+// Special block that will have ULID of the meta.json being referenced too.
+const blockIDLabel = "__block_id"
+
+// Filter filters out blocks that have no labels after relabelling of each block external (Thanos) labels.
 func (f *LabelShardedMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta, synced GaugeLabeled, _ bool) {
+	var lbls labels.Labels
 	for id, m := range metas {
-		if processedLabels := relabel.Process(labels.FromMap(m.Thanos.Labels), f.relabelConfig...); processedLabels != nil {
-			continue
+		lbls = lbls[:0]
+		if len(lbls) < len(m.Thanos.Labels)+1 {
```

Comment:
you reset the length in the line above, I assume you meant to do `cap` on the left side here?